crypto/x509/x_attrib.c

```diff
@@ -159,10 +159,13 @@ int ossl_print_attribute_value(BIO *out,
     switch (av->type) {
     case V_ASN1_BOOLEAN:
         if (av->value.boolean) {
-            return BIO_printf(out, "%*sTRUE", indent, "");
+            if (BIO_printf(out, "%*sTRUE", indent, "") < 0)
+                return 0;
         } else {
-            return BIO_printf(out, "%*sFALSE", indent, "");
+            if (BIO_printf(out, "%*sFALSE", indent, "") < 0)
```

Comment:
Should `<= 0` be used here? That's what is used in the V_ASN1_INTEGER case. Although it probably doesn't matter much I'd like to at least see that the condition for the BIO_printf result is the same everywhere in this function.

Reply:
Good catch. I fixed this (but haven't pushed it yet). I was actually thinking about the changes I made in this PR like this. I did a lot of work to transform the `BIO_` functions return codes into `0` upon failure, or `1` upon success. In hindsight, I am thinking maybe I should have just changed `ossl_print_attribute_value()` to have return codes in line with these. What do you think about this?

Reply:
Not sure. You mean returning the return value from BIO_printf()? I do not think the number of bytes printed is not that useful.

Reply:
Well, I just figured I could say "0 or greater is a success code and less than 0 is a failure," then I could just return the values of `BIO_printf()` directly. I am not trying to preserve the number of bytes printed as much as I am trying to simplify the code.

Reply:
If you want to simplify the code then do: `return BIO_printf(...) >= 0;`

Reply:
Wonderful! That was a great suggestion. I have fixed this. I also fixed a bug where this would have returned an error code if `indent` was set to `0` (which doesn't happen anywhere in the code now, but it could in the future.)